### PR TITLE
Add support for memo field in add funds modal

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/6049

⚠️ Require: https://github.com/opencollective/opencollective-api/pull/8081

[memo-field.webm](https://user-images.githubusercontent.com/12435965/196364827-4cf02bbc-bb37-49d4-8df6-8a6029b2fce4.webm)

### CODE REVIEW

File: components/host-dashboard/AddFundsModal.js [hunk 1]
```diff
@@ -500,6 +505,14 @@ const AddFundsModal = ({ collective, ...props }) => {
                   >
                     {({ field }) => <StyledInput data-cy="add-funds-description" {...field} />}
                   </StyledInputFormikField>
+                  <StyledInputFormikField
+                    name="memo"
+                    htmlFor="addFunds-memo"
+                    label={<FormattedMessage defaultMessage="Memo (Optional)" />}
```
Comment (hdiniz) [inline components/host-dashboard/AddFundsModal.js:511]: Using required = {false} already adds a optional hint.
```suggestion
                    required={false}
                    label={<FormattedMessage defaultMessage="Memo" />}
```

image

**Comment by znarf** [comment]
Let's have a tooltip to explain what's a "Memo" please.

File: components/host-dashboard/AddFundsModal.js (hunk 1 above)
Comment (SudharakaP) [reply components/host-dashboard/AddFundsModal.js]: Thanks. Missed this. 😉

**Comment by SudharakaP** [comment]
> Let's have a tooltip to explain what's a "Memo" please.

Added a tooltip which says, `This is a private note that will only be visible to the host.`

**Review by hdiniz — APPROVED** [review]
LGTM

**Pull request merged**